Line/iOS/LINELoginDemo/LINELoginDemo/AppDelegate.m

```diff
@@ -0,0 +1,97 @@
+    [self validateConfigs];
+    
+    // Initialize SDKs
+    [LineAdapter handleLaunchOptions:launchOptions];
+    [FIRApp configure];
+    
+    return YES;
+}
+
+
+- (void)applicationWillResignActive:(UIApplication *)application {
+    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+    // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
+}
+
+
+- (void)applicationDidEnterBackground:(UIApplication *)application {
+    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+}
+
+
+- (void)applicationWillEnterForeground:(UIApplication *)application {
+    // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+}
+
+
+- (void)applicationDidBecomeActive:(UIApplication *)application {
+    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+}
+
+
+- (void)applicationWillTerminate:(UIApplication *)application {
+    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+}
+
+- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation
+{
+    return [LineAdapter handleOpenURL:url];
+}
+
+- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url
+{
+    return [LineAdapter handleOpenURL:url];
+}
+
+- (void)validateConfigs {
+    // Check if LINE SDK has been setup
+    NSString *linePlistPath = [[NSBundle mainBundle] pathForResource:@"LineAdapter" ofType:@"plist"];
+    NSDictionary *linePlist = [[NSDictionary alloc] initWithContentsOfFile:linePlistPath];
+    BOOL isLineSetupFinished = ![linePlist[@"ChannelId"] isEqualToString:@"<your_channel_id>"];
+    NSAssert(isLineSetupFinished, @"Please update your LINE Channel ID in LineAdapter.plist");
+    
+    // Check if Firebase SDK has been setup
+    NSString *googleServicesPlistPath = [[NSBundle mainBundle] pathForResource:@"GoogleService-Info" ofType:@"plist"];
+    BOOL isFirebaseSetupFinished = googleServicesPlistPath != nil;
+    NSAssert(isFirebaseSetupFinished, @"Please download GoogleService-Info.plist from Firebase Console and add to your project.");
```

Comment:
This check for GoogleService-info isn't needed. FIRApp config already raises an exception if it can't find that file.